ddtrace/tracer/transport.go

```diff
@@ -183,6 +186,24 @@ func (t *httpTransport) endpoint() string {
 	return t.traceURL
 }
 
+// getAgentURL determines the trace agent URL. The value held by environment variable
+// DD_TRACE_AGENT_URL takes precedence over other configuration options.
+func getAgentURL(addr string) string {
+	if agentURL := os.Getenv("DD_TRACE_AGENT_URL"); agentURL != "" {
+		u, err := url.Parse(agentURL)
+		if err != nil {
+			log.Warn("Failed to parse DD_TRACE_AGENT_URL: %v", err)
+			return fmt.Sprintf("http://%s", resolveAgentAddr(addr))
+		}
+		if u.Scheme != "http" && u.Scheme != "https" && u.Scheme != "unix" {
```

Comment:
```suggestion
		if u.Scheme != "http" {
```
What's the point of allowing any type of `unix://` scheme or elsewhere, if `http://` is used anyway?

Reply:
I could be wrong, but my read is that this code checks if the scheme in the provided URL is http, https, or unix. If it is none of those things, it will warn and then default to http. If it *is* any of those things, this code block is never entered, and the agent URL, as provided by the env var, is used, unmodified.

Reply:
Now "unix" is treated as separate, because if the customer sets a unix domain socket in DD_TRACE_AGENT_URL the http client's Transport needs to be modified. "http" and "https" should be supported in latest changes as well. 
Let me know what you think!